src/Microsoft.TemplateEngine.Edge/Template/WellKnownSearchFilters.cs

```diff
@@ -181,5 +181,31 @@ public static class WellKnownSearchFilters
                 return null;
             };
         }
+
+        public static Func<ITemplateInfo, MatchInfo?> AuthorFilter(string author)
+        {
+            return (template) =>
+            {
+                if (string.IsNullOrEmpty(author) || string.IsNullOrWhiteSpace(template.Author))
+                {
+                    return null;
+                }
+
+                int authorIndex = template.Author.IndexOf(author, StringComparison.OrdinalIgnoreCase);
+
+                if (authorIndex == 0 && string.Equals(template.Author, author, StringComparison.OrdinalIgnoreCase))
```

Comment:
Instead of calling `string.Equals(...)` again here, wouldn't length comparison be sufficient?